ICSSoft.STORMNET.Business/SQLDataService/SQLDataService.Async.cs

```diff
@@ -0,0 +1,402 @@
+                object[][] result = string.IsNullOrEmpty(selectString) ? new object[0][] : await ReadAsync(
+                    selectString,
+                    0).ConfigureAwait(false);
+
+                ConvertReadResult(result, dataObjects, customizationStruct, storageStruct, allObjectKeys, readingKeys, clearDataObject, dataObjectCache);
+            }
+            finally
+            {
+                dataObjectCache.StopCaching();
+            }
+        }
+
+        /// <inheritdoc cref="IAsyncDataService.LoadObjectsAsync(LoadingCustomizationStruct, DataObjectCache)"/>
+        public virtual async Task<DataObject[]> LoadObjectsAsync(LoadingCustomizationStruct customizationStruct, DataObjectCache dataObjectCache = null)
+        {
+            if (dataObjectCache == null)
+            {
+                dataObjectCache = new DataObjectCache();
+            }
+
+            using (DbTransactionWrapperAsync wrapper = new DbTransactionWrapperAsync(this))
+            {
+                return await LoadObjectsByExtConnAsync(customizationStruct, dataObjectCache, wrapper)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Асинхронная загрузка объектов с использованием указанной коннекции и транзакции.
+        /// </summary>
+        /// <param name="customizationStruct">Структура, определяющая, что и как грузить.</param>
+        /// <param name="dataObjectCache">Кэш объектов для зачитки.</param>
+        /// <param name="dbTransactionWrapperAsync">Коннекция и транзакция, через которые будут выполнена зачитка.</param>
+        /// <returns>Загруженные данные.</returns>
+        public virtual async Task<DataObject[]> LoadObjectsByExtConnAsync(
+            LoadingCustomizationStruct customizationStruct,
+            DataObjectCache dataObjectCache,
+            DbTransactionWrapperAsync dbTransactionWrapperAsync)
+        {
+            if (customizationStruct == null)
+            {
+                throw new ArgumentNullException(nameof(customizationStruct));
+            }
+
+            if (dbTransactionWrapperAsync == null)
+            {
+                throw new ArgumentNullException(nameof(dbTransactionWrapperAsync));
+            }
+
+            if (dataObjectCache == null)
+            {
+                dataObjectCache = new DataObjectCache();
+            }
+
+            dataObjectCache.StartCaching(false);
+            try
+            {
+                RunChangeCustomizationString(customizationStruct.LoadingTypes);
+
+                // Применим полномочия на строки.
+                ApplyReadPermissions(customizationStruct, SecurityManager);
+
+                Type[] dataObjectType = customizationStruct.LoadingTypes;
+                StorageStructForView[] storageStruct;
+
+                string selectString = string.Empty;
+                selectString = GenerateSQLSelect(customizationStruct, false, out storageStruct, false);
```

Comment:
Объединить с декларацией.
```suggestion
                string selectString = GenerateSQLSelect(customizationStruct, false, out storageStruct, false);
```

Reply:
Сделано